simple-chat-back/src/utils/users.js

```diff
@@ -0,0 +1,40 @@
+const users = []
+
+const addUser = ({id, username, room}) => {
+    username = username.trim().toLowerCase()
+    room = room.trim().toLowerCase()
+
+    if(!username || !room){
+        return{
+            error: 'Username and room are required!'
```

Comment:
Don't assume your message errors are going to be explicitly presented on the frontend like the comment before. A frontend will only check the message and then show a personalized one, on this case please make it lowercase and remove the exclamation mark